lib/credentials/shared_ini_file_credentials.js

```diff
@@ -143,17 +151,11 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
       );
 
       if (profile['role_arn'] && !preferStaticCredentialsToRoleArn) {
-        this.loadRoleProfile(profiles, profile, function(err, data) {
-          if (err) {
-            callback(err);
-          } else {
+        this.loadRoleProfile(profiles, profile, function(err) {
+          if (!err) {
             self.expired = false;
-            self.accessKeyId = data.Credentials.AccessKeyId;
-            self.secretAccessKey = data.Credentials.SecretAccessKey;
-            self.sessionToken = data.Credentials.SessionToken;
-            self.expireTime = data.Credentials.Expiration;
```

Comment:
These properties now get assigned in `STSCredentialsCache ` via `credentialsFrom`.